packages/pds/src/pipethrough.ts

```diff
@@ -179,32 +334,111 @@ export const parseProxyHeader = async (
   return { did, serviceUrl }
 }
 
-// Sending request
+// Request parsing/forwarding
 // -------------------
 
-export const makeRequest = async (
-  url: URL,
-  reqInit: RequestInit,
-): Promise<Response> => {
-  let res: Response
+type Accept = [name: string, flags: { q: string }]
```

Comment:
thoughts on 

```
type Accept = {
  name: string
  q: string
}
```

instead of the array syntax? seems more clear & type-safe

Reply:
The idea of this code is to be generic of the `accept-*` header being negotiated.

Consider for example:

```
Accept-Patch: text/example;charset=utf-8, text/example
```

In that case, the `Accept` type could be generalized as such:
```ts
type Accept = [name: string, flags: Record<string, string>]
```

If you think this is too much YAGNI I can make this more straight forward.

What type safety difference are you thinking of ?